packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -70,175 +74,321 @@ fn get_pkg_path(name: &str, pkg_type: PackageType) -> String {
-                typeof start_experiment === 'undefined' ? undefined : start_experiment,
-                typeof start_sim === 'undefined' ? undefined : start_sim,
-                typeof run_task === 'undefined' ? undefined : run_task
-            ]}})",
-            code
-        );
-        let pkg: mv8::Function<'_> = mv8
-            .eval(wrapped_code)
-            .map_err(|e| Error::PackageImport(path.clone(), e.into()))?;
-        let args = mv8::Values::from_vec(vec![
-            embedded.hash_util.clone(),
-            embedded.hash_stdlib.clone(),
-        ]);
-
-        let fns: mv8::Array<'_> = pkg
-            .call(args)
-            .map_err(|e| Error::PackageImport(path.clone(), e.into()))?;
-        if fns.len() != 3 {
-            return Err(Error::PackageImport(path.clone(), "Stray return".into()));
+        // Double braces like `{{` are Rust's escape string for a single literal `{`.
+        let wrapped_code = new_js_string(
+            scope,
+            &format!(
+                "((hash_util, hash_stdlib) => {{{code}
+                    return [
+                        typeof start_experiment === 'undefined' ? undefined : start_experiment,
+                        typeof start_sim === 'undefined' ? undefined : start_sim,
+                        typeof run_task === 'undefined' ? undefined : run_task
+                    ]
+                }})",
+            ),
+        )?;
+
+        let pkg: v8::Local<'_, v8::Function> = {
+            let mut try_catch = v8::TryCatch::new(scope);
+
+            let pkg = v8::Script::compile(&mut try_catch, wrapped_code, None).ok_or_else(|| {
+                let exception = try_catch.exception().unwrap();
+                let exception_string = exception
+                    .to_string(&mut try_catch)
+                    .unwrap()
+                    .to_rust_string_lossy(&mut try_catch);
+
+                Error::PackageImport(
+                    path.clone(),
+                    format!("Could not compile code for package: {exception_string}"),
+                )
+            })?;
+
+            let pkg = pkg.run(&mut try_catch).ok_or_else(|| {
+                let exception = try_catch.exception().unwrap();
+                let exception_string = exception
+                    .to_string(&mut try_catch)
+                    .unwrap()
+                    .to_rust_string_lossy(&mut try_catch);
+
+                Error::PackageImport(
+                    path.clone(),
+                    format!("Couldn't execute package setup: {exception_string}"),
+                )
+            })?;
+
+            pkg.try_into().map_err(|err| {
+                Error::PackageImport(
+                    path.clone(),
+                    format!("Could not convert package setup return value into Function: {err}"),
+                )
+            })?
+        };
+
+        let args = &[embedded.hash_util, embedded.hash_stdlib];
+        let fns: v8::Local<'_, v8::Array> = {
```

Comment:
Can we add an `Array` type as well?

Reply:
Added in https://github.com/hashintel/hash/pull/414/commits/85562cb5e340eb009f3768779dcde15bdee9ed7a